cmd/server/shared/shared.go

```diff
@@ -194,17 +210,59 @@ func Main() {
 		procDiedAction = goreman.Ignore
 	}
 
-	// If in restore mode, only run PostgreSQL
-	if restore, _ := strconv.ParseBool(os.Getenv("PGRESTORE")); restore {
-		procfile = []string{}
-		procfile = append(procfile, postgresLine)
+	runMigrations := !restore
+	run(procfile, postgresProcfile, runMigrations, procDiedAction)
+}
+
+func run(procfile, postgresProcfile []string, runMigrations bool, procDiedAction goreman.ProcDiedAction) {
+	if !runMigrations {
+		procfile = append(procfile, postgresProcfile...)
+		postgresProcfile = nil
 	}
 
-	err = goreman.Start([]byte(strings.Join(procfile, "\n")), goreman.Options{
+	group, _ := errgroup.WithContext(context.Background())
+
+	options := goreman.Options{
 		RPCAddr:        "127.0.0.1:5005",
 		ProcDiedAction: procDiedAction,
-	})
-	if err != nil {
+	}
+	startProcesses(group, "postgres", postgresProcfile, options)
+
+	if runMigrations {
+		// Run migrations before starting up the application but after
+		// starting any postgres instance within the server container.
+		runMigrator()
+	}
+
+	startProcesses(group, "all", procfile, options)
+
+	if err := group.Wait(); err != nil {
 		log.Fatal(err)
 	}
 }
+
+func startProcesses(group *errgroup.Group, name string, procfile []string, options goreman.Options) {
+	if len(procfile) == 0 {
+		return
+	}
+
+	log.Printf("Starting %s processes\n", name)
```

Comment:
nit: you don't need newlines for log pkg
```suggestion
	log.Printf("Starting %s processes", name)
```